src/mongocxx/validation_criteria.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Class representing criteria for document validation, to be applied to a collection.
+///
+class MONGOCXX_API validation_criteria {
+   public:
+    ///
+    /// Set a validation rule for this validation object.
+    ///
+    /// @param rule
+    ///   Document representing a validation rule.
+    ///
+    /// TODO: add a link to documentation once it is published.
```

Comment:
Depending on which documentation, this would probably go better on line 30 (the comment for the class)